piff/gp_interp.py

```diff
@@ -0,0 +1,324 @@
+                 logger=None):
+        from sklearn.gaussian_process import GaussianProcessRegressor
+        if isinstance(kernel, str):
+            kernel = self._eval_kernel(kernel)
+
+        if attr_interp is None:
+            attr_interp = ['u', 'v']
+
+        self.attr_interp = attr_interp
+        self.kernel = kernel
+        self.npca = npca
+
+        self.kwargs = {
+            'attr_interp': attr_interp,
+            'optimizer': optimizer,
+            'npca': npca,
+            'kernel': repr(kernel)
+        }
+
+        self.gp = GaussianProcessRegressor(self.kernel, optimizer=optimizer)
+
+    @staticmethod
+    def _eval_kernel(kernel):
+        from sklearn.gaussian_process.kernels import WhiteKernel, RBF, Matern, RationalQuadratic
+        from sklearn.gaussian_process.kernels import ExpSineSquared, DotProduct, PairwiseKernel
+        from numpy import array
+        return eval(kernel)
+
+    def _fit(self, X, y, logger=None):
+        """Update the GaussianProcessRegressor with data
+        :param X:  The independent covariates.  (n_samples, n_features)
+        :param y:  The dependent responses.  (n_samples, n_targets)
+        """
+        # Save these for potential read/write.
+        self._X = X
+        self._y = y
+        if self.npca > 0:
+            from sklearn.decomposition import PCA
+            self._pca = PCA(n_components=self.npca, whiten=True)
+            self._pca.fit(y)
+            y = self._pca.transform(y)
+        self.gp.fit(X, y)
+
+    def _predict(self, Xstar):
+        """ Predict responses given covariates.
+        :param X:  The independent covariates at which to interpolate.  (n_samples, n_features).
+        :returns:  Regressed parameters  (n_samples, n_targets)
+        """
+        ystar = self.gp.predict(Xstar)
+        if self.npca > 0:
+            ystar = self._pca.inverse_transform(ystar)
+        return ystar
+
+    def getProperties(self, star, logger=None):
+        """Extract the appropriate properties to use as the independent variables for the
+        interpolation.
+
+        Take self.attr_interp from star.data
+
+        :param star:    A Star instances from which to extract the properties to use.
+
+        :returns:       A np vector of these properties.
+        """
+        return np.array([star.data[key] for key in self.attr_interp])
+
+    def initialize(self, stars, logger=None):
+        """Initialize both the interpolator to some state prefatory to any solve iterations and
+        initialize the stars for use with this interpolator.
+
+        :param stars:   A list of Star instances to interpolate between
+        :param logger:  A logger object for logging debug info. [default: None]
+        """
+        self.solve(stars, logger=logger)
+        return self.interpolateList(stars)
+
+    def solve(self, stars=None, logger=None):
+        """Set up the GaussianProcessRegressor to be able to predict.
```

Comment:
GaussianProcessRegressor is an implementation detail.  Since `solve` is a public API method, this should be reworded.  Probably just replace with `this GPInterp object`.